### ISSUE TITLE
SPAKE2+ support

### ISSUE THREAD

**Issue by est31** [RustCrypto/PAKEs#30]
I see SPAKE2+ mentioned in the cargo docs of the SPAKE crate. But it's not implemented yet, is it? It would be cool to have it.

**Comment by warner** [comment]
Nope, not yet.. sounds like a good idea to me too. We'll need to extend the API to provide a registration step, since the symmetric PAKEs don't have one.

**Comment by tarcieri** [comment]
Nope, the `spake2` crate is largely in maintenance mode, but contributions would be welcome

**Comment by potto216** [comment]
If PRs are still being accepted on this issue I would like to contribute since it is useful for Matter home automation implementations. Also, if PRs are being accepted, I noticed several PRs such as https://github.com/RustCrypto/PAKEs/pull/153 and https://github.com/RustCrypto/PAKEs/pull/153 have been reviewed but haven't been merged. Is there a general sticking point that I should be aware of when doing the PR? Thanks

**Comment by tarcieri** [comment]
@potto216 it would be great if you can take a look. This repo unfortunately doesn't get the attention some others get, most notably because none of the current maintainers are experts in PAKE algorithms including SPAKE2.

Regarding the other PRs, they were all abandoned by their authors with open review questions.

[This list of questions](https://github.com/RustCrypto/PAKEs/pull/166#issuecomment-2198193493) is still relevant.

**Comment by potto216** [comment]
My initial thoughts on the architecture are to add `pub struct Spake2Plus` to `spake2/src/lib.rs` and implement functions such as `start_prover` and `start_verifier` to use naming consistent with [RFC 9383](https://www.rfc-editor.org/rfc/rfc9383.html) similar to the A/B used in the code from [RFC 9382](https://www.rfc-editor.org/rfc/rfc9382.html).  Also functions for transcript generation need to be added to spake2/src/ed25519.rs.
